### ISSUE TITLE
Have METexpress better handle weird EMC data

### ISSUE THREAD

**Issue by mollybsmith-noaa** [dtcenter/METexpress#152]
Recently, someone loaded some METplus data to the EMC AWS database that had region names containing a ".". Region names that I've come across have always been dot-free previously, which is important because regions are used as keys in the apps' MongoDB instance, and MongoDB keys can't have dots in them. Therefore, we need to assume that this will happen again and sanitize anything coming from the metadata that might be used as a key. Previously we've done this by string replacing dots with "____" (four underbars), which, at least so far, no one has attempted to use in the name of any METplus fields. We can probably use a similar method again.

**Comment by ian-noaa** [comment]
It looks like MongoDB has historically had issues with `$` and `.` characters. It sounds like Mongo 5.0 supports `$` and `.` but there are some gotchas so it seems like it'd still be worthwhile to have a "`sanitizeForMongo()`" (or whatever makes sense naming-wise) method to rewrite them both, while we're at it.

Previously, Mongo had recommended rewriting them to the unicode character equivalent:

* U+FF04 (i.e. `＄`)
* U+FF0E (i.e. `．`)

I'm not sure if they're relevant but there are some other naming limits in databases, & selection names as well. https://www.mongodb.com/docs/manual/reference/limits/#naming-restrictions

**Comment by mollybsmith-noaa** [comment]
Yeah, I have a list of characters that Mongo doesn’t like, I just never
thought anyone would actually use one of them in a region name. Oh well,
now I get to sanitize all of the metadata instead of just some of it.

**Comment by mollybsmith-noaa** [comment]
This is done, METexpress now replaces "." with "___" for any strings used as keys, and then replaces them back in the data routines.
